docs_src/design/adr/0020-UnitOfMeasure.md

```diff
@@ -0,0 +1,78 @@
+
+There are various specifications and standards around unit of measure.  Specifically, there are several options to choose from as it relates to the exchange of data in electronic communications - and units of measure associated in that exchange.  As examples, two big competing standards around EDI (electronic data exchange) that both have associated unit of measure codes are:
+
+- [ANSI X12: EDI standard used mostly in the US](https://ediacademy.com/blog/x12-unit-of-measurement-codes/)
+- [EDIFACT: UN EDI standard used mostly in Europe and Asia](https://unece.org/fileadmin/DAM/cefact/recommendations/rec20/rec20_rev3_Annex2e.pdf)
+
+The [Unified Code for Units of Measure](https://en.wikipedia.org/wiki/Unified_Code_for_Units_of_Measure) provides an alternative list (not a standard) that is used by various organizations like OSGI and the Eclipse Foundatin.
+
+While standards exist, use by various open source projects (especially IoT/edge projects) is inconsistent and haphazard.   Groups like oneM2M seem to define their own selection of units in specifications per vertical (home for example) while Kura doesn't even appear to use the UoM JSR (a Java related unit of measure specification for Java applications like Kura).
+
+## Decision
+It would be speculative and inappropriate for EdgeX to select a unit of measure which is not widely adopted in the industry or choose a static unit of measure list that is incomplete with regard to possible IoT / edge use case needs.
+
+Therefore, EdgeX chooses not to select or adopt a unit of measure specification, standard, or code list to apply across the platform.  Instead, EdgeX adopters will be allowed to optionally specify which unit of measure specification, standard, or unit of measure code list they would like applied and understood to be represented by values in EdgeX.
+
+### Specifying Unit of Measure per Device Resource
+Going forward, an EdgeX property will be optionally associated to each device resource to stipulate which unit of measure standard (or specification, code list, etc.) is applied to the unit of measure of the device resource.
+
+For example, if the device resource for temperature was specified in a device profile as show below, the unit of measure would be specified by the units section (specifying the Cel as the unit of measure from the Unified Code for Units of Measure).
+
+``` YAML
+-
+  name: "RoomTemperature"
+  isHidden: false
+  description: "Room Temperature x10 °C (Read Only)"
+  attributes:
+    { primaryTable: "INPUT_REGISTERS", startingAddress: 3 }
+  properties:
+    valueType: "Float32"
+    readWrite: "R"
+    scale: "1"
+  units:
+    value:  "Cel"
+    standard: "UCUM"
```

Comment:
currently, units is a pure string.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/352324e9c8b8d76ffd6147dc5ec2da6dd8f275fd/v2/dtos/resourceproperties.go#L15

for the backward compatibility, we can just add a new field called units_standard, not changing the units data type.
```
  units: "Cel"
  units_standard: "UCUM"
```